stargate/stargate
# closes #1717: boostratping the Docs API V2 with Quarkus

**What this PR does**:

This PR provides the Quarkus bootstrapping for the Docs API V2.

The best way to review this is to first read the `README.MD` file, I know it's long, but once you read it, it will be easier to understand all the code and configuration. Some major concerns are solved here in addition to the stuff provided by the Quarkus itself. Those are not related to the Document API, but to new micro services in general:

* header-based security
* multi-tenancy
* Cassandra token resolving
* gRPC metadata passing

There is an `ExampleResource` that explains how the flow should look like.

The PR does not contain any document related code. Note that only `.java` files, `pom.xml`, `application.yaml`, banner config and `README.MD` files were added by me. Everything rest is generated and I guess should not be reviewed.

It's important to mention that this project is independent for now, it's not part of the parent project. We need to find a way to have projects with different Java target in the mono-repo. Quarkus supports only Java 11+ and I bootstrapped with Java 17 actually. And used some nice new features as well. I guess the right way to do this would be [toolchains](https://maven.apache.org/guides/mini/guide-using-toolchains.html). Or optionally breaking out from the main repository..
 
Code is not formatted, and CI is not checking this project for now. All of this will be done, once we connect to the parent project. The observability parts (metrics, tracing) are also not included as we have a separate issue for this -> #1719.

The majority of this code should go to some kind of `svg2-common-quarkus` project, but there is no need to do this, until we have another project moving to Quarkus. 

The project for now depends only on the `grpc-proto`, we will connect it do the existing `sgv2-service-common` project once we need features from it.

I ll present more in-depth details on my demo on Thursday. Unit tests for all new components will be added once we agree on the approach being valid.

**Which issue(s) this PR fixes**:
Relates to #1717.

**Checklist**
- [x] Changes manually tested
- [x] Finalize OpenAPI dev props 
- [x] Solve formatting, the `fmt-maven-plugin` does not support Java 17
- [x] Integrate into the CI so tests are verified
- [x] Quarkus update to `2.8.0.Final`
    - check support for `@Nested` fixed
    - check support for `MockitoExtension` fixed
